src/Options.php

```diff
@@ -9,29 +9,67 @@ class Options
      *
      * @var array
      */
-    public $fileTypes = [".php"];
+    private $fileTypes = [".php"];
 
     /**
-     * @param \stdClass|null $options
+     * @param \Traversable|\stdClass|array|null $options
      */
-    public function __construct(\stdClass $options = null)
+    public function __construct($options = null)
     {
         // Do nothing when the $options parameter is not an object
-        if (!is_object($options)) {
+        if (!is_object($options) && !is_array($options) && (!$options instanceof \Traversable)) {
             return;
         }
 
-        $this->fileTypes = $options->fileTypes ?? $this->normalizeFileTypes($this->fileTypes);
+        foreach ($options as $option => $value) {
+            $method = 'set' . ucfirst($option);
+
+            call_user_func([$this, $method], $value);
+        }
+    }
+
+    /**
+     * Validate and set options for file types
+     *
+     * @param array $fileTypes List of file types
+     */
+    public function setFileTypes(array $fileTypes)
+    {
+        $fileTypes = filter_var_array($fileTypes, FILTER_SANITIZE_STRING);
+        $fileTypes =  filter_var($fileTypes, FILTER_CALLBACK, ['options' => [$this, 'filterFileTypes']]);
+        $fileTypes = array_filter($fileTypes);
```

Comment:
https://secure.php.net/manual/en/function.array-filter.php says, that the second parameter must be provided. But you could use `strlen`, that will filter any false, empty strings or null.